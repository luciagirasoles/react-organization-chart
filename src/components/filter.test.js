import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Filter from "./filter";

test("Filter", () => {
  const onChange = jest.fn();
  const { getByLabelText, asFragment } = render(
    <Filter titles={["CEO", "Testing Engineer"]} value="" onChange={onChange} />
  );

  expect(asFragment()).toMatchSnapshot();

  const select = getByLabelText("Select a filter");

  fireEvent.change(select, { target: { value: "CEO" } });

  expect(onChange).toHaveBeenCalledWith("CEO");
});
