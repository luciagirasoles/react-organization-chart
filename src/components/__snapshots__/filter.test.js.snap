// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter 1`] = `
<DocumentFragment>
  .emotion-4 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  padding: .5rem;
  margin-bottom: 1rem;
  position: absolute;
  display: inline-block;
  width: auto;
  top: 1rem;
  right: 1rem;
}

@media (max-width:460px) {
  .emotion-4 {
    position: static;
    width: 100%;
  }
}

@media (max-width:460px) {
  .emotion-0 {
    display: block;
    margin: 1rem 0;
  }
}

.emotion-3 {
  -webkit-appearance: none;
  color: rgb(0,0,0);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 40px;
  font-size: 12px;
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 100;
  position: relative;
  white-space: nowrap;
  line-height: 0;
  width: 300px;
  min-width: 160px;
  background: rgb(255,255,255);
  outline: none;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(234,234,234);
  border-image: initial;
  overflow: hidden;
  -webkit-transition: border 0.2s ease 0s,background 0.2s ease 0s,color 0.2s ease-out 0s,box-shadow 0.2s ease 0s;
  transition: border 0.2s ease 0s,background 0.2s ease 0s,color 0.2s ease-out 0s,box-shadow 0.2s ease 0s;
  border-radius: 4px;
}

@media (max-width:460px) {
  .emotion-3 {
    width: 100%;
  }
}

.emotion-1 {
  height: 100%;
  box-shadow: none;
  color: rgb(0,0,0);
  line-height: 40;
  font-size: 14px;
  margin-right: -20px;
  width: calc(100% + 20px);
  text-transform: none;
  border-width: initial;
  border-style: none;
  border-color: initial;
  border-image: initial;
  background: none transparent;
  padding: 0px 76px 0px 16px;
  outline: none;
}

.emotion-2 {
  width: 30px;
  height: 100%;
  position: absolute;
  right: 0;
  pointer-events: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-left: 1px solid rgb(234,234,234);
  background: rgb(255,255,255);
  -webkit-transition: border 0.2s ease 0s;
  transition: border 0.2s ease 0s;
}

<div
    class="emotion-4"
  >
    <form>
      <label
        class="emotion-0"
        for="filter"
      >
        Highlight employees with title: 
      </label>
      <div
        class="emotion-3"
      >
        <select
          aria-label="Select a filter"
          class="emotion-1"
          id="filter"
          name="filter"
        >
          <option
            value=""
          >
            -- Select an title --
          </option>
          <option
            value="CEO"
          >
            CEO
          </option>
          <option
            value="Testing Engineer"
          >
            Testing Engineer
          </option>
        </select>
        <div
          class="emotion-2"
        >
          <svg
            aria-label="arrow double"
            fill="none"
            height="17"
            viewBox="0 0 7 12"
            width="7"
          >
            <path
              d="M0.642491 3.35053L0.292945 3.70804L1.00798 4.40714L1.35752 4.04962L0.642491 3.35053ZM3.75752 1.59491L4.10707 1.23739L3.39204 0.538299L3.04249 0.895815L3.75752 1.59491ZM5.58506 4.04651L5.93149 4.40704L6.65256 3.71417L6.30613 3.35364L5.58506 4.04651ZM3.95354 0.9053L3.6071 0.544767L2.88604 1.23763L3.23247 1.59817L3.95354 0.9053ZM1.35752 7.95041L1.00797 7.59289L0.292938 8.29198L0.642485 8.6495L1.35752 7.95041ZM3.04248 11.1042L3.39203 11.4617L4.10706 10.7626L3.75751 10.4051L3.04248 11.1042ZM6.36054 8.64636L6.70697 8.28583L5.98591 7.59296L5.63947 7.95349L6.36054 8.64636ZM3.28688 10.4018L2.94045 10.7624L3.66152 11.4552L4.00795 11.0947L3.28688 10.4018ZM1.35752 4.04962L3.75752 1.59491L3.04249 0.895815L0.642491 3.35053L1.35752 4.04962ZM6.30613 3.35364L3.95354 0.9053L3.23247 1.59817L5.58506 4.04651L6.30613 3.35364ZM0.642485 8.6495L3.04248 11.1042L3.75751 10.4051L1.35752 7.95041L0.642485 8.6495ZM5.63947 7.95349L3.28688 10.4018L4.00795 11.0947L6.36054 8.64636L5.63947 7.95349Z"
              fill="#000"
            />
          </svg>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
